docs/ref/signals.txt

```diff
@@ -142,6 +142,21 @@ Arguments sent with this signal:
     The set of fields to update as passed to :meth:`.Model.save`, or ``None``
     if ``update_fields`` wasn't passed to ``save()``.
 
+Alternative to using ``pre_save``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If the model you are targeting is part of your project, a more robust and maintainable
+solution would be to override the model's :meth:`~django.db.models.Model.save` method::
+
+    class Article(models.Model):
+        ...
+        def save(self, update_fields=None, **kwargs):
+            self.slug = slugify(self.title)
+            if update_fields:
+                update_fields = set(update_fields) | {"slug"}
+
+            super().save(update_fields=update_fields, **kwargs)
+
```

Comment:
I think these examples might be a bit much. Do you think we can just link back to the docs that already cover this stuff: 

```
.. warning::

    Consider more robust and maintainable solutions before using signals. See  
    :ref:`overriding-model-methods` for some examples. 
```

This would work for the pre/post delete/save signals, and the section there already mentions the bulk delete question and such.